pkg/gitfile/scmprovider.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2022 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitfile
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+var (
+	invalidSourceError = errors.New("invalid source string")
+)
+
+// ScmProvider defines the interface that in order to determine if URL belongs to SCM provider
+type ScmProvider interface {
```

Comment:
These guys should probably be somehow linked or derived from the service providers in the main operator codebase, IMHO. But maybe we can do that at some later point in time. But since we're already starting to look at gitlab, we should probably do it soone than later. Because the gitlab impl here only support gitlab.com which I'm sure is not going to be the only gitlab server we're going to be connecting to.

Reply:
Agree, but that's another huge piece of refactoring, i would do that separately.